### PR TITLE
Improve Split block.

### PR DESCRIPTION
**Pull request by MidgeOnGithub**
The splitter could not split when transitioning into level 19 (Meteorite Cavern) because this if block was slightly malformed. As it was, it also evaluated the "end of game" logic when the player was just entering level 19. Now, it only evaluates the "end of game" logic if the player has actually been playing level 19.

### CODE REVIEW

File: LiveSplit.TR3.asl [hunk 1]
```diff
@@ -132,7 +132,7 @@ start
     }
     else  // IL-specific logic
     {
-        bool goingFromOneLevelToAnother = old.level != current.level && current.currentLevelTime == 0;
+        bool goingFromOneLevelToAnother = old.level != current.level && current.currentLevelTime == 0 && !current.isTitle;
```
Comment (rtrger) [inline LiveSplit.TR3.asl:135]: `isTitle` is not needed here.

File: LiveSplit.TR3.asl [hunk 2]
```diff
@@ -144,19 +144,32 @@ reset
 
 split
 {
-    // Determine if a split should occur.
     bool shouldSplit;
-    if (current.level == 19)
+    if (current.level == 19 && old.level == 19)
+    {
+        // While playing the "last" level (regardless of IL/FG and if the bonus level, All Hallows, will be played)
```
Comment (rtrger) [inline LiveSplit.TR3.asl:150]: Imho this is a bit too verbose: "Last level: split when player's control ends and helicopter scene begins." should suffice.

File: LiveSplit.TR3.asl [hunk 3]
```diff
@@ -144,19 +144,32 @@ reset
...
+    {
+        // An easy way to check if the player has advanced into the next level is by checking currentLevelTime == 0.
+        // Because it is impossible to pause and make a save within the first few frames of a level, this check
+        // will not fail if the player happens to loads a save into the next level.
         shouldSplit = current.currentLevelTime == 0 && old.currentLevelTime > 0;
+    }
 
     var index = vars.completedLevels.IndexOf(current.level);
-    bool levelWasAlreadyCompleted = index != -1;
+    bool levelWasAlreadyCompleted = index != -1;  // A result of -1 would indicate current.level was not found.
```
Comment (rtrger) [inline LiveSplit.TR3.asl:170]: I think the comment is not needed, since this can be found in .NET's documentation.

File: LiveSplit.TR3.asl [hunk 4]
```diff
@@ -144,19 +144,32 @@ reset
...
 split
 {
-    // Determine if a split should occur.
     bool shouldSplit;
-    if (current.level == 19)
+    if (current.level == 19 && old.level == 19)
+    {
+        // While playing the "last" level (regardless of IL/FG and if the bonus level, All Hallows, will be played)
+        // the timer should split when the player's control ends and the helicopter scene begins.
+        // This occurs when the player hits the end level trigger which sets levelComplete to true.
```
Comment (rtrger) [inline LiveSplit.TR3.asl:152]: Imho this is a bit too verbose: "Last level: split when player's control ends and helicopter scene begins." should suffice.

File: LiveSplit.TR3.asl [hunk 5]
```diff
@@ -144,19 +144,32 @@ reset
...
         shouldSplit = current.levelComplete && !old.levelComplete;
+    }
     else if (settings["IL"])
+    {
+        // For IL timing, the timer should stop when the end-level stats screen appears.
+        // Checking levelComplete ensures this check works in all cases.
         shouldSplit = current.levelComplete && current.isStatsScreen && !old.isStatsScreen;
```
Comment (rtrger) [inline LiveSplit.TR3.asl:159]: To do: try this without `current.levelComplete`, if works, then I think the comments are not needed.

File: LiveSplit.TR3.asl [hunk 6]
```diff
@@ -144,19 +144,32 @@ reset
...
+    }
     else
+    {
+        // An easy way to check if the player has advanced into the next level is by checking currentLevelTime == 0.
+        // Because it is impossible to pause and make a save within the first few frames of a level, this check
+        // will not fail if the player happens to loads a save into the next level.
```
Comment (rtrger) [inline LiveSplit.TR3.asl:165]: This else block = Not the last level and not IL: split when timer is reset to 0.

I'd delete the comment.

File: LiveSplit.TR3.asl (hunk 5 above)
Comment (MidgeOnGithub) [reply LiveSplit.TR3.asl:156]: I did not try without the levelComplete check but I removed the comment.

File: LiveSplit.TR3.asl (hunk 6 above)
Comment (MidgeOnGithub) [reply LiveSplit.TR3.asl]: I cut this explanation down to two lines; I believe the explanation about saves and timer could be beneficial. I'm not opposed to a complete removal, though.

File: LiveSplit.TR3.asl (hunk 4 above)
Comment (MidgeOnGithub) [reply LiveSplit.TR3.asl]: I reduced the wordiness of this.

**Review by rtrger — APPROVED** [review]

**Pull request merged**